WordPress/Classes/ViewRelated/NUX/SiteCreationFields.swift

```diff
@@ -0,0 +1,64 @@
+import Foundation
+
+/// Singleton class to contain options selected by the user
+/// during the Site Creation process.
+///
+class SiteCreationFields {
+
+    // MARK: - Properties
+
+    private static var privateShared: SiteCreationFields?
+
+    var title = ""
+    var tagline: String?
+    var theme: Theme?
+    var domain = ""
+
+    /// An enum for returning validation Errors.
+    private enum SiteCreationFieldsError: Error {
+        case missingTitle
+        case missingDomain
+        case domainContainsWordPressDotCom
+        case missingTheme
+    }
+
+    // MARK: - Init
+
+    private init() {
+    }
+
+    // MARK: - Instance Methods
+
+    static func sharedInstance() -> SiteCreationFields {
+        if privateShared == nil {
+            privateShared = SiteCreationFields()
+        }
+        return privateShared!
+    }
+
+    static func resetFields() {
```

Comment:
I could be wrong, but `resetSharedInstance()` feels like a more accurate method name? `resetFields()` sounds like it'd be an instance method?